src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -1,11 +1,422 @@
+        String[] lessonDetails = input.trim().split(" ", 3);
+
+        // split the details field using DELIMITER to get the individual detail fields
+        String[] details = lessonDetails[2].split(DELIM);
+
+        // store detail fields that have been filled by the user into an array
+        // if user did not enter that field, it will remain as an empty string
+        for (int i = 0; i < details.length && i < allDetails.length; i++) {
+            allDetails[i] = details[i].trim();
+        }
+
+        // Creating Lesson Object
+        String type = allDetails[LESSON_TYPE_INDEX].toUpperCase();
+        // throws "illegal argument exception" if enum value is wrong
+        LessonType TYPE = LessonType.valueOf(type);
+        String timeAndDay = allDetails[LESSON_TIME_DAY_INDEX];
+        String link = allDetails[LESSON_LINK_INDEX];
+        String teacherName = allDetails[LESSON_TEACHER_INDEX];
+        String email = allDetails[LESSON_EMAIL_INDEX];
+
+        TeachingStaff teacher = new TeachingStaff(teacherName, email);
+
+        return new Lesson(TYPE, timeAndDay, link, teacher);
+    }
+
+    /**
+     * Parses details of new task from user input string.
+     *
+     * @param input full user input string
+     * @return a Task object with the details entered by the user. Fields not found will be left as an empty string.
+     */
+    private Task parseNewTaskDetails(String input) {
+
+        // initialize an array of empty strings to store task details
+        String[] allDetails = new String[TASK_MAX_DETAILS];
+        Arrays.fill(allDetails, "");
+
+        // to remove only the first two words "add task"
+        String[] taskDetails = input.trim().split(" ", 3);
+
+        // split the details field using DELIMITER to get the individual detail fields
+        String[] details = taskDetails[2].split(DELIM);
+
+        // store detail fields that have been filled by the user into an array
+        // if user did not enter that field, it will remain as an empty string
+        for (int i = 0; i < details.length && i < allDetails.length; i++) {
+            allDetails[i] = details[i].trim();
+        }
+
+        // Creating Task object
+        String description = allDetails[TASK_DESCRIPTION_INDEX];
+        String deadlineString = allDetails[TASK_DEADLINE_INDEX];
+        LocalDate deadline = convertToDate(deadlineString);
+        String remarks = allDetails[TASK_REMARKS_INDEX];
+
+        return new Task(description, deadline, remarks);
+    }
+
+    /**
+     * Converts given string to LocalDate object.
+     *
+     * @param string string to be converted
+     * @return LocalDate object of the date represented by the string
+     * @throws DateTimeParseException if invalid input format / invalid date given
+     */
+    private LocalDate convertToDate(String string) throws DateTimeParseException {
+        DateTimeFormatter parseFormat = DateTimeFormatter.ofPattern("d-M-yyyy");
```

Comment:
Pattern can be declared as a string constant.

Reply:
Oops, will fix